File: scala/src/main/scala/com/authdog/exceptions/ApiException.scala
package com.authdog.exceptions

/**
 * Exception thrown when API requests fail
 */
class ApiException(message: String = "API request failed", cause: Option[Throwable] = None) 
  extends AuthdogException(message, cause)

object ApiException {
  def apply(message: String): ApiException = new ApiException(message)
  def apply(message: String, cause: Throwable): ApiException = new ApiException(message, Some(cause))
}

File: java/src/main/java/com/authdog/exceptions/ApiException.java
package com.authdog.exceptions;

/**
 * Exception thrown when API requests fail.
 */
public class ApiException extends AuthdogException {

    /**
     * Constructor with message.
     * @param messageParam Error message
     */
    public ApiException(final String messageParam) {
        super(messageParam);
    }

    /**
     * Constructor with message and cause.
     * @param messageParam Error message
     * @param causeParam Cause of the exception
     */
    public ApiException(final String messageParam, final Throwable causeParam) {
        super(messageParam, causeParam);
    }
}
